testing/agent-exporter/src/main/java/io/opentelemetry/javaagent/testing/provider/AgentTestExporterCustomizerProvider.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.testing.provider;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.javaagent.testing.exporter.AgentTestingExporterFactory;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.DeclarativeConfigurationCustomizer;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.DeclarativeConfigurationCustomizerProvider;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.ConsoleExporterModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.LogRecordExporterModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.LogRecordProcessorModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.LoggerProviderModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.MeterProviderModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.MetricReaderModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.OpenTelemetryConfigurationModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.PeriodicMetricReaderModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.PushMetricExporterModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.SimpleLogRecordProcessorModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.SimpleSpanProcessorModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.SpanExporterModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.SpanProcessorModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.TracerProviderModel;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@AutoService(DeclarativeConfigurationCustomizerProvider.class)
+public class AgentTestExporterCustomizerProvider
+    implements DeclarativeConfigurationCustomizerProvider {
+  @Override
+  public void customize(DeclarativeConfigurationCustomizer customizer) {
+    AgentTestingExporterFactory.init();
+
+    customizer.addModelCustomizer(
+        model -> {
+          if (model.getTracerProvider() == null) {
+            addTracerProvider(model);
+          }
+          if (model.getLoggerProvider() == null) {
+            addLoggerProvider(model);
+          }
+          if (model.getMeterProvider() == null) {
+            addMeterProvider(model);
+          }
+          return model;
+        });
+  }
+
+  private static void addTracerProvider(OpenTelemetryConfigurationModel model) {
```

Comment:
perhaps it would help if there was a comment that shows what is added. For example I think this adds
```
tracer_provider:
  processors:
    - simple:
        exporter:
          agent_test:
    - simple:
        exporter:
          console:
```